cni/pkg/nodeagent/ztunnelserver.go

```diff
@@ -59,37 +60,39 @@ To clean up stale ztunnels
 */
 
 type connMgr struct {
-	connectionSet []*ZtunnelConnection
+	connectionSet []ZtunnelConnection
 	mu            sync.Mutex
 }
 
-func (c *connMgr) addConn(conn *ZtunnelConnection) {
+func (c *connMgr) addConn(conn ZtunnelConnection) {
+	log := log.WithLabels("conn_uuid", conn.uuid)
 	log.Info("ztunnel connected")
 	c.mu.Lock()
 	defer c.mu.Unlock()
 	c.connectionSet = append(c.connectionSet, conn)
 	ztunnelConnected.RecordInt(int64(len(c.connectionSet)))
 }
 
-func (c *connMgr) LatestConn() *ZtunnelConnection {
+func (c *connMgr) LatestConn() (ZtunnelConnection, error) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 	if len(c.connectionSet) == 0 {
-		return nil
+		return ZtunnelConnection{}, fmt.Errorf("no connection")
 	}
-	return c.connectionSet[len(c.connectionSet)-1]
+	return c.connectionSet[len(c.connectionSet)-1], nil
 }
 
-func (c *connMgr) deleteConn(conn *ZtunnelConnection) {
+func (c *connMgr) deleteConn(conn ZtunnelConnection) {
+	log := log.WithLabels("conn_uuid", conn.uuid)
 	log.Info("ztunnel disconnected")
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
-	// Loop over the slice, keeping non-deleted conn pointers
-	// but filtering out the deleted one.
-	var retainedConns []*ZtunnelConnection
+	// Loop over the slice, keeping non-deleted conn but
+	// filtering out the deleted one.
+	var retainedConns []ZtunnelConnection
 	for _, existingConn := range c.connectionSet {
-		// Not conn-by-pointer that was deleted? Keep it.
+		// Not conn that was deleted? Keep it.
 		if existingConn != conn {
```

Comment:
So we are still comparing pointers implicitly here by `u` and `Update`. Not sure if this is fine or not but it seems like the behavior here should ultimately be the same as before?

Reply:
Yes, those are still pointers but the UUID being part of the struct means the equality test will fail if the pointers are  the same but the struct isn't, which mimics the behavior pre-https://github.com/istio/istio/pull/54565/files#diff-9873062ab6cdec1c9e1829c5cad625c4ef9d878a05b8b96ee3449623ea50f442L64